shared/actions/chat/index.js

```diff
@@ -301,19 +307,23 @@ function* _updateThread({
   let newMessages = []
   const newUnboxeds = (thread && thread.messages) || []
   for (const unboxed of newUnboxeds) {
-    const message = _unboxedToMessage(unboxed, yourName, yourDeviceName, conversationIDKey)
+    const message: Constants.ServerMessage = _unboxedToMessage(
+      unboxed,
+      yourName,
+      yourDeviceName,
+      conversationIDKey
+    )
     const messageFromYou =
       message.deviceName === yourDeviceName && message.author && yourName === message.author
 
-    let pendingMessage
     if ((message.type === 'Text' || message.type === 'Attachment') && messageFromYou && message.outboxID) {
-      pendingMessage = yield select(Shared.messageOutboxIDSelector, conversationIDKey, message.outboxID)
-    }
-
-    if (pendingMessage) {
-      // Delete the pre-existing pending version of this message, since we're
-      // about to add a newly received version of the same message.
-      yield put(Creators.removeOutboxMessage(conversationIDKey, message.outboxID))
+      const outboxID: Constants.OutboxIDKey = message.outboxID
+      const pendingMessage = yield select(Shared.messageOutboxIDSelector, conversationIDKey, outboxID)
```

Comment:
fyi yield select returns any. @chrisnojima and I think it's best to do `yield select()` and then do Shared.messageOutboxIDSelector(state, conversationIDKey, outboxID)

Reply:
Done, and fixed another instance in the diff. There's a lot more in the file, but I don't want this PR to sprawl...